### ISSUE TITLE
The `opensAnAddress` reader welds the name of a call over an escaped backslash and a digit, and reads what an address holds

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#579]
Found on the branch that closes #526, whose reading is the mirror of this one.

### What minimal example or steps are needed to reproduce the bug?

An address standing behind a word that ends in an escaped backslash and a digit:

```css
a { b: x\\9 url(1PX); }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json
```

### What did you expect to happen?

Nothing. `x\\9` is an identifier ending in an escaped backslash and a digit, the space behind it is the text's, and `url(1PX)` is an address, whose text is nobody's to recase: `@csstools/css-tokenizer` reads an ident token, whitespace and a url token, and `lightningcss@1.33.0` prints the line as `a { b: x\\9 url("1PX"); }`, quoting the address and keeping its letters.

### What actually happened?

```
1:18-20  Expected "PX" to be "px"
```

and `--fix` leaves `a { b: x\\9 url(1px); }`, the address rewritten.

The same happens to every rule that asks `opensAnAddress` whether a call is one:

| input | rule and option | what `--fix` leaves |
| --- | --- | --- |
| `a { b: x\\9 url(1PX); }` | `unit-case: lower` | `a { b: x\\9 url(1px); }` |
| `a { b: x\\61 url(1PX); }` | `unit-case: lower` | `a { b: x\\61 url(1px); }` |
| `a { b: \\9 url(#FFF); }` | `color-hex-case: lower` | `a { b: \\9 url(#fff); }` |
| `a { b: x\\9 url(.5); }` | `number-leading-zero: always` | `a { b: x\\9 url(0.5); }` |

### What is the cause?

`readName` in `lib/utils/opensAnAddress/index.ts` gathers the name of a call back over the whitespace a hexadecimal escape swallows — the reading #344 asked for, so that `\61 url(` names the call `aurl` — and asks `TRAILING_HEX_ESCAPE` whether the word in front of the space ends in such an escape. That expression matches a backslash and hexadecimal digits at the end of the word and counts no backslashes in front of them, so `x\\9` answers yes: the escaped backslash and the digit are taken for the escape `\9`, the space for its closing character, and the name is read as `x\\9 url`, which names no address. The word `x\9`, whose backslash is one, does weld the name — `x\9 url(` is the function token `x\9 url(` to the tokenizer, and `lightningcss` recases the `1PX` inside it — so the two spellings part exactly at the parity of the run, which is what `spelledRuns` reads and this reader does not.

### Do you have a proposal to fix the bug?

Ask `spelledRuns` whether the last run of the word is an escape that is a backslash and hexadecimal digits alone, the way the branch closing #526 asks it when welding the words such an escape parted, and let `readName` and that welding share the one reading.
